util/encoding/encoding_test.go

```diff
@@ -346,18 +348,71 @@ func TestEncodeDecodeUvarint(t *testing.T) {
 
 // TestDecodeInvalidUvarint tests that invalid uvarint encodings panic.
 func TestDecodeInvalidUvarint(t *testing.T) {
-	testCases := [][]byte{
-		// length of 9 bytes should cause an error.
-		{0x11, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01},
+	tests := []struct{
+		name    string // name printed with errors.
```

Comment:
You could also combine the tests by making the decode function a parameter of the struct:

```
  tests := []struct {
    ...
    decode func([]byte)
  }{
    {
      ...
      decode: func(b []byte) { DecodeUvarint(b) },
    },
  }
```